js/private/js_run_devserver.mjs

```diff
@@ -402,14 +459,25 @@ async function main(args, sandbox) {
                         console.error('IBAZEL_BUILD_COMPLETED SUCCESS')
                     }
                     // Chain promises via syncing.then()
-                    syncing = syncing.then(() =>
-                        sync(
-                            // Re-parse the config file to get the latest list of data files to copy
-                            JSON.parse(fs.readFileSync(configPath)).data_files,
+                    syncing = syncing.then(() => {
+                        // Re-parse the config file to get the latest list of data files to copy
+                        const updatedDataFiles = JSON.parse(
+                            fs.readFileSync(configPath)
+                        ).data_files
+                        // Remove files that were previously synced but are no longer in the updated list of files to sync
+                        deleteFiles(
+                            config.data_files,
+                            updatedDataFiles,
+                            sandbox
+                        )
+                        // Sync changed files
+                        config.data_files = updatedDataFiles
```

Comment:
We don't want to overwrite the full `config`? Just overwrite the `data_files` within it? 🤷

Reply:
I don't want to get into test cases for every possible thing that could change if the user starts changing the devserver target. Would rather have users re-start the ibazel process if they are changing something fundamental about the devserver target besides the graph.